root/etc/e-smith/templates/etc/ssh/sshd_config/45AllowGroups2Sshd

```diff
@@ -0,0 +1,54 @@
+{
+    # Allow groups to sftp/ssh with different policies
+    # legacy: everybody with a ssh account can login
+    # admins: groups of $admins{'group'} can login
+    # custom:  group listed $ssh{'AllowGroups'} can login
+    # members restricted to sftp (SftpRestrictedGroups) must be allowed to sshd also
+
+    use strict;
+    use esmith::ConfigDB;
+    use esmith::AccountsDB;
+    my $db = esmith::ConfigDB->open_ro();
+
+
+    sub uniq {
+        my %seen;
+        return grep { !$seen{$_}++ } @_;
+    }
+
+    my $domain = $db->get_value('DomainName')|| die ('Cannot retrieve DomainName');
+    my $PermitRootLogin = $db->get_prop('sshd','PermitRootLogin') || "no";
+    my $policy = $db->get_prop('sshd','Policy') || "legacy";
+
+    my @AllowGroups = ();
+    foreach ( split(',',$db->get_prop('sshd','AllowGroups') || '')) {
+        my ($group, $sftp) = split(':', $_);
+
+        if(!$group || !$sftp) {
+            next;
+        }
+        push @AllowGroups, $group;
+    }
+
+    # ssh doesn't accept to login if we restrict with short group name
+    # It seems to want the long group name group@domain.com
```

Comment:
Probably it is not a real issue but it's better to know... what about the letter case? `Domain Admins` vs `domain admins`. Is it also a case-sensitive match?

Reply:
sorry I did not catch you, could elaborate I cannot find it in the code

Reply:
Test case:
- create a group called MixedCaseGroup
- add members to it and add it to `AllowGroups`
- check SSH access works for members of MixedCaseGroup

Reply:
our validator inside cockpit doesn't accept capital letter for group creation

Reply:
same for nethgui, capital letter are not accepted

Reply:
Good to know, that means the corner is even more... corner! There is still the (small) chance of getting capital letters from a remote LDAP or AD (non-nethserver).

Reply:
I tested with ssh{AllowGroups}, `Domain Admins` is accepted (service not broken) and the user member of `domain admins` is refused, We have a tiny chance to get it workable :D

Reply:
Ok, but if nethserver is bound to a third-party remote accounts provider and the group name is "Domain Admins", I'd expect also `AllowGroups=Domain Admins`, or consistently `admins/group=Domain Admins`. Right?

Reply:
For my understanding `Domain Admins` != `domain admins` but if you want we could lowercase all the group name, but I am not sure it is good